addons/binding/org.openhab.binding.s7/src/main/java/org/openhab/binding/s7/handler/S7BridgeHandler.java

```diff
@@ -0,0 +1,342 @@
+        }
+    }
+
+    private void destroyClient() {
+        if (client != null) {
+            client.Disconnect();
+            client = null;
+        }
+    }
+
+    private void createPollingJob() {
+        logger.debug("Creating new polling job for S7 server {} with interval {}ms.", getConfig().get(HOST),
+                getConfig().get(POLLING_INTERVAL));
+        int pollingInterval = DEFAULT_POLLING_INTERVAL;
+        try {
+            Object pollingIntervalConfig = getConfig().get(POLLING_INTERVAL);
+            if (pollingIntervalConfig != null) {
+                pollingInterval = ((BigDecimal) pollingIntervalConfig).intValue();
+            } else {
+                logger.debug("Polling interval not configured for this S7 client. Using default value: {}s",
+                        pollingInterval);
+            }
+        } catch (NumberFormatException ex) {
+            logger.info("Wrong configuration value for polling interval. Using default value: {}s", pollingInterval);
+        }
+
+        pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, 5000, pollingInterval, TimeUnit.MILLISECONDS);
+    }
+
+    private Hashtable<Integer, byte[]> ReadData() {
+        final int PALength = 64;
+        final int PELength = 64;
+        final int DBLength = 96;
+        final int MKLength = 32;
+
+        Hashtable<Integer, byte[]> data = new Hashtable<Integer, byte[]>();
+
+        byte[] dataPA = new byte[PALength];
+        byte[] dataPE = new byte[PELength];
+        byte[] dataDB = new byte[DBLength];
+        byte[] dataMK = new byte[MKLength];
+
+        synchronized (client) {
+            if (client.ReadArea(S7.S7AreaPA, 1, 0, PALength, dataPA) == 0) {
+                data.put(S7.S7AreaPA, dataPA);
+                nReadError = 0;
+            } else {
+                nReadError++;
+            }
+
+            if (client.ReadArea(S7.S7AreaPE, 1, 0, PELength, dataPE) == 0) {
+                data.put(S7.S7AreaPE, dataPE);
+                nReadError = 0;
+            } else {
+                nReadError++;
+            }
+
+            if (client.ReadArea(S7.S7AreaDB, 1, 0, DBLength, dataDB) == 0) {
+                data.put(S7.S7AreaDB, dataDB);
+                nReadError = 0;
+            } else {
+                nReadError++;
+            }
+
+            if (client.ReadArea(S7.S7AreaMK, 1, 0, MKLength, dataMK) == 0) {
+                data.put(S7.S7AreaMK, dataMK);
+                nReadError = 0;
+            } else {
+                nReadError++;
+            }
+        }
+
+        if (nReadError == 10) {
+            onConnectionLost();
+        }
+
+        return data;
+    }
+
+    public void setBit(int Area, int Address, boolean value) {
+        byte Data[] = new byte[1];
+
+        synchronized (client) {
+            client.ReadArea(Area, 1, Address / 8, 1, Data);
+            S7.SetBitAt(Data, 0, Address % 8, value);
+            client.WriteArea(Area, 1, Address / 8, 1, Data);
```

Comment:
Just for info: Moka7 version in PLCLogo binding was patched by me to be able to write single bit.